tests/bluetooth/tester/src/l2cap.c

```diff
@@ -404,6 +415,31 @@ static void listen(uint8_t *data, uint16_t len)
 		   BTP_STATUS_FAILED);
 }
 
+static void credits(uint8_t *data, uint16_t len)
+{
+	const struct l2cap_credits_cmd *cmd = (void *) data;
+	struct channel *chan = &channels[cmd->chan_id];
+
+	if (!chan->in_use) {
+		goto fail;
+	}
+
+	if (chan->nbuf && bt_l2cap_chan_recv_complete(&chan->le.chan,
+							chan->nbuf) < 0) {
+		goto fail;
+	}
+
+	chan->nbuf = NULL;
```

Comment:
How does the reference counting of this buffer work? I don't seen any usage of `net_buf_ref()` or `net_buf_unref()`?

Reply:
My understanding was that returning -EINPROGRESS in recv callback give ownership to user and thus extra reffing is not needed cause it will be returned back to host via bt_l2cap_chan_recv_complete() call

I can add reffing it it is required but from quick look at shell sample it seems it is not needed.

Reply:
No need to add if it's all as intended, but I'd at least add code comments to explain why this is correct behaviour. What about the case when bt_l2cap_chan_recv_complete() fails and `chan->nbuf` remains set? When will it be freed?

Reply:
hmm right, I'll clean it in disconnected callback